pyccl/halos/profiles/profile_base.py

```diff
@@ -52,12 +53,36 @@ def __init__(self, *, mass_def=None, concentration=None):
         # Initialize FFTLog.
         self.precision_fftlog = FFTLogParams()
 
+        if (mass_def, concentration) == (None, None):
+            warnings.warn(
+                "mass_def (or concentration where applicable) will become a "
+                "required argument for HaloProfile instantiation in CCLv3 "
+                "and will be moved from (real, fourier, projected, cumul2d "
+                "convergence, shear, reduced_shear, magnification).",
+                CCLDeprecationWarning)
+            self.mass_def = self.concentration = None
+            return
+
         # Initialize mass_def and concentration.
         self.mass_def, *out = MassDef.from_specs(
             mass_def, concentration=concentration)
         if out:
             self.concentration = out[0]
 
+        # TODO: Remove for CCLv3.
+        self._is_number_counts = isinstance(self, HaloProfileNumberCounts)
```

Comment:
Humm, but this doesn't happen if `mass_def` is None, right?

Reply:
indeed, good catch

Reply:
done